pkg/operator/operator.go

```diff
@@ -735,7 +775,31 @@ func (o *Operator) sync(ctx context.Context, key string) error {
 		klog.Warningf("Fail to load ConsoleConfig, AlertManager's externalURL may be outdated")
 	}
 
-	factory := manifests.NewFactory(o.namespace, o.namespaceUserWorkload, config, o.loadInfrastructureConfig(ctx), proxyConfig, o.assets, apiServerConfig, consoleConfig)
+	// Enable kube-state-metrics' custom-resource-state-based metrics if VPA CRD is installed within the cluster.
+	enableKSMCRSMetrics := false
+	isVPACRDPresent, err := o.client.VPACustomResourceDefinitionPresent(ctx, o.lastKnownVPACustomResourceDefinitionPresent)
+	if err != nil {
+		// Throw on all transient errors.
+		return fmt.Errorf("unable to guess the desired state for kube-state-metrics' custom-resource-state metrics enablement: %w", err)
+	} else {
+		// If we didn't get an error, we can safely assume that the CRD is deterministically either present or absent.
+		if *isVPACRDPresent {
```

Comment:
Nit: Do we really need `enableKSMCRSMetrics` here? If `isVPACRDPresent != nil` update `o.lastKnownVPACustomResourceDefinitionPresent` and if its even true, log the message.